src/dftbp/dftb/scc.F90

```diff
@@ -526,20 +542,64 @@ subroutine getAtomicGammaMatrix(this, gammamat, iNeighbour, img2CentCell)
     !> index array between images and central cell
     integer, intent(in) :: img2CentCell(:)
 
+    integer :: iam, nprocs
+
     @:ASSERT(this%tInitialised)
     @:ASSERT(all(shape(gammamat) == [ this%nAtom, this%nAtom ]))
     @:ASSERT(this%elstatType == elstatTypes%gammaFunc)
 
   #:if WITH_SCALAPACK
+    
     call error("scc:getAtomicGammaMatrix does not work with MPI yet")
+    
   #:endif
-
+    
     gammamat(:,:) = this%coulomb%invRMat
     call this%shortGamma%addAtomicMatrix(gammamat, iNeighbour, img2CentCell)
 
   end subroutine getAtomicGammaMatrix
 
+#:if WITH_SCALAPACK
+  
+  subroutine getAtomicGammaMatrixBlacs(this, gammamat, iNeighbour, img2CentCell, env)
+
+    !> Instance
+    class(TScc), intent(in) :: this
+
+    !> Environment settings
+    type(TEnvironment), intent(in) :: env
+
+    !> Atom resolved gamma
+    real(dp), intent(out) :: gammamat(:,:)
+
+    !> neighbours of atoms
+    integer, intent(in) :: iNeighbour(0:,:)
+
+    !> index array between images and central cell
+    integer, intent(in) :: img2CentCell(:)
+
+    integer :: ii, jj, iLoc, jLoc, rSrc, cSrc
 
+    @:ASSERT(this%tInitialised)
+    @:ASSERT(all(shape(gammamat) == [ this%nAtom, this%nAtom ]))
+    @:ASSERT(this%elstatType == elstatTypes%gammaFunc)
+
+    gammamat(:,:) = 0.0_dp
+    do ii = 1, this%coulomb%descInvRMat_(M_)
```

Comment:
Is this copying the distributed matrix to a local matrix?  The current structure behaves like a serial part of the code, as every process loops over the global matrix sizing.The gemr2d routine might be faster, or alternatively, loop over the local size of the invRMat and use scalafx_indxl2g to find the global location (see calls in coulomb.F90).

Reply:
Yes, this is to get a global copy of Gamma. The partitioning of RPA vectors is not done with PBLAS, hence there is no point in distributing gamma. Can you come up with a nicer way to build it? I tried a lot of different things. This solution worked at least.